### ISSUE TITLE
Configs of Adam for training ViT

### ISSUE THREAD

**Issue by zhengmeixi** [Ardcy/Lookaround#2]
Hi, thanks for your job and for sharing the code.

I'm trying to reproduce the result in table3 about finetuning ViT on cifar10 with SWA(98.47% reported). The  paper appendix C.2 said the adam weight decay should be set as 0.01: "For the training of ViT-B/16, we used the Adam optimizer, starting with a learning rate of 0.00001. We set β1 to 0.9, β2 to 0.999, and applied a weight decay of 0.01." However, in code train_vit_cifar10_pretrain.yaml, the adam weight decay is set as 0.00005. Which value should I use?

**Comment by zhengmeixi** [comment]
To reproduce ViT-B/16 SWA CIFAR-10 in table 3(98.47%), I use the following yaml file
```
MODEL:
  NAME: 'vit'
  CLASSES: 10
  DEVICE: 'cuda'
  PRETRAINED: True

DATA_DIR: 'dataset'

DATASET:
  IMAGESIZE: 224
  NAME: 'cifar10'

SOLVER:
  OPTIMIZER_NAME: 'Adam'
  Adam_LR: 0.001
  Adam_Beta1: 0.9
  Adam_Beta2: 0.999
  Adam_weight_decay: 0.00005
  LR: 0.1
  BATCH_SIZE: 64
  SCHEDULER: 'Cosine'
  SWA_LR: 0.0008
```
with script
`--yaml_path train_vit_cifar10_pretrain.yaml --train_mode TRAIN_SWA --cuda_id 0 --data_dir /data/cifar10 --out ./out/ --optimizer Adam`

To reproduce ViT-B/16 Lookaround CIFAR-10 in table 3(98.71%), I use the following yaml file
```
MODEL:
  NAME: 'vit'
  CLASSES: 10
  DEVICE: 'cuda'
  PRETRAINED: True

DATA_DIR: 'dataset'

DATASET:
  IMAGESIZE: 224
  NAME: 'cifar10'

SOLVER:
  OPTIMIZER_NAME: 'Adam'
  Adam_LR: 0.001
  Adam_Beta1: 0.9
  Adam_Beta2: 0.999
  Adam_weight_decay: 0.00005  # these adam optimizer config seems wouldn't work and will be covered by Lookaround optimizer
  LR: 0.1
  BATCH_SIZE: 64
  SCHEDULER: 'Cosine'
```
with script
--yaml_path train_vit_cifar10_pretrain.yaml --train_mode TRAIN_LOOKAROUND --cuda_id 0 --data_dir /data/cifar10 --out ./out/ --optimizer Lookaround

Can you help me check whether I use the correct configs? Thanks

**Comment by Ardcy** [comment]
You should set the weight decay to 0.01. In the initial experiments, I set it to 0.00005 and got poor results. Later, I corrected it. Thank you for pointing that out.

**Comment by zhengmeixi** [comment]
Thanks for your reply!! Can you also help me check whether other configs I used are correct, especially **lr**(I comment a moment ago). Thanks!!!

**Comment by Ardcy** [comment]
The correct learning rate should be 0.00001. The description in our paper is correct.

An additional suggestion is to choose a larger batch size (e.g., 128, 256). The results might be better.

**Comment by zhengmeixi** [comment]
So, to finetune ViT with lookaround on CIFAR10, I should set the scheduler  as cosine annealing scheduler whose initial lr is 0.00001 and T_max is 200 epochs, right?
And I also notice the swa_lr is set as 0.0008 in your codes, is this value too small for swa? Cause the learning rate for swa is typically 0.05 or 0.1.

**Comment by Ardcy** [comment]
For the first question, "So, to finetune ViT with lookaround on CIFAR10, I should set the scheduler as cosine annealing scheduler whose initial lr is 0.00001 and T_max is 200 epochs, right?"

Yes, your understanding is right.

For the second question, "And I also notice the swa_lr is set as 0.0008 in your codes, is this value too small for swa? Cause the learning rate for swa is typically 0.05 or 0.1."

Actually, I think the swa_lr should be 0.0008. Because the 0.0008 doesn't work at the beginning of training, but only at the final stage.

You need to note that SWA method only be adopted at the final stages of training. During other epochs, the training process of SWA and SGD/Adam is the same, and the learning rate between both is the same too. So, at the final stage, the learning rate has already been reduced significantly, and switching to a higher learning rate at this point can cause a significant drop in accuracy.

**Comment by zhengmeixi** [comment]
Thanks a lot!! I'll try the configs you suggest :)
